File: Leetcode 3536 Maximum Product of Two Digits.py
# 2026/08/06
class Solution:
    def maxProduct(self, n: int) -> int:
        dict_n = {
            1:0,
            2:0,
            3:0,
            4:0,
            5:0,
            6:0,
            7:0,
            8:0,
            9:0,
            0:0,
        }
        for i in str(n):
            dict_n[int(i)] += 1
        dict_n = {k: dict_n[k] for k in sorted(dict_n, reverse=1) if dict_n[k] != 0}
        ans = 1
        muli = 0 # max 2
        for key, item in dict_n.items():
            if item > 1 and muli == 0:
                return key*key
            else:
                ans *= key
                muli += 1
            if muli == 2:
                break
        return ans

        

File: Leetcode 3536 Maximum Product of Two Digits.go
// 2026/08/06

func maxProduct(n int) int {
    a := [10]int{}
    
    for _, i := range strconv.Itoa(n) {
        s := int(i - '0')
        a[s] += 1
    }
    slices.Reverse(a[:])
    ans := 1
    count := 0 
    for index, num := range(a){
        if num > 1 && count == 0 {
            return (9-index)*(9-index)
        } else if num > 0 {
            ans *= (9-index)
            count += 1
        }
        if count == 2 {
            break
        }
    }
    if count < 2 {
        return 0
    }

    return ans
}
